engine/server/engine/enclave_manager/enclave_pool.go

```diff
@@ -62,17 +62,28 @@ func CreateEnclavePool(
 		)
 	}
 
-	// The amount of idle enclaves is equal to the chan capacity + one enclave (this one will be waiting in the queue until the channel is unblocked)
-	// chanCapacity = 0  means that it will be an unbuffered, so the communication between the channels will succeed when both (sender and receiver) are ready
-	chanCapacity := poolSize - oneEnclave
+	// this channel is the repository of idle enclave UUIDs
+	idleEnclavesChan := make(chan enclave.EnclaveUUID, poolSize)
+
+	// this channel is for communicate to the sub-routine that one
+	// idle enclave has been picked from the pool, so it can trigger the
+	// fill mechanism again, it has the capacity = poolSize for not blocking
+	// the caller
+	fillChan := make(chan bool, poolSize)
 
-	idleEnclavesChan := make(chan enclave.EnclaveUUID, chanCapacity)
+	// this channel if for signaling the close event to the sub-routine
+	closeChan := make(chan bool)
+
+	ctxWithCancel, cancelCtxFunc := context.WithCancel(context.Background())
```

Comment:
Using a cancel context is clever here, but then it raises the question of why you need the closeChan, aren't those two redundant?

Reply:
yes they are, good catch!